types/react/index.d.ts

```diff
@@ -483,7 +492,13 @@ declare namespace React {
         altKey: boolean;
         charCode: number;
         ctrlKey: boolean;
+        /**
+         * See [DOM Level 3 Events spec](https://www.w3.org/TR/uievents-key/#keys-modifier). for a list of valid (case-sensitive) arguments to this method.
```

Comment:
You could also change the definition to take a union of string literal types to describe the set of valid arguments.

Reply:
I would, but there are some none standard values that are useful in some browsers in some circumstances. I could try to list those too but I feel that the situation is that we then are trying to specify higher type safety than there is. All strings are allowed even though only a subset is usable which is similar to the situation with array indices, real numbers are allowed, just not useful. I also think many valid uses of it would error due to literal type widening as it is likely that you for example have an array of modifiers that you check.